Card: Lacy O'Malley, Zeb Whateley-Dupont

mmeldo opened the pull request:
Updated `moveCard` player function (created a new one to take also context) so it can be checked if the moving card is done due to card ability.
Updated effects that uses minimal bullets (cannot be less than).

mmeldo commented:
Help with testing needed:
- Maggie Harris
- Harold Aimslee
- Devil's Six Gun
or some other card that puts a card from discard pile to play.

stopfstedt commented:
Testing Maggie's Job - this works as intended. Job goes through, horse is selected but doesn't enter play. However, the message output could use some improvement. perhaps an additional `Lacy O'Malley prevents {0} from entering play.` would be helpful to the players here.

![image](https://user-images.githubusercontent.com/1410427/130710330-5e3307c4-2ab8-425b-913f-23222bec1b6b.png)

stopfstedt commented:
Devil's Six Gun works as intended - search for Joker is triggered, but doesn't enter hand. The "some effect prevents" message output is spot on, something like that should be added to Maggie's output.

![image](https://user-images.githubusercontent.com/1410427/130710679-c5cd2878-93e0-4678-85a4-038e185ab113.png)

stopfstedt commented:
Testing with Harold Aimslee - works as intended! 

![image](https://user-images.githubusercontent.com/1410427/130710957-092d5400-4da4-477c-a5ee-14101e603b63.png)

stopfstedt reviewed (changes requested):
Needs additional message output if the target of Maggie Harris' job resolution was prevented from entering play. LGTM otherwise.

mmeldo commented:
> Needs additional message output if the target of Maggie Harris' job resolution was prevented from entering play. LGTM otherwise.

When updating the `PutIntoPlay` action to show message that the target (horse) was prevented from entering play, I found another problem. Problem was that the `PutIntoPlay` action could not be checked if it was a card ability so I fixed that.
Another thing is that there is a lot more cards similar to Maggie that needed updating. And a special chapter was `Dr. Dawn Edwards` vs `Eve Henry`. It was not correctly implemented before so I fixed it.

To test this, please test at least 3 other cards putting stuff into play from discard. For example Pete Spence, Land Purchase and Recruitment Drive along with Dr. Dawn Edwards and Eve Henry combo. I will test some other cards.

stopfstedt commented:
> To test this, please test at least 3 other cards putting stuff into play from discard. For example Pete Spence, Land Purchase and Recruitment Drive along with Dr. Dawn Edwards and Eve Henry combo. I will test some other cards.

re-testing tonight.
